### PR TITLE
rad-toa product for sentinel2

### PR DESCRIPTION
**Pull request by ags-tolson**
I want to get feedback on the algorithm I'm using to generate rad-toa and the design choices I made.  See especially `process_rad` in `sentinel2Data`, and `radiance_factors` in `sentinel2Asset`

This code has successfully produced a valid, but not necessarily correct, rad-toa tif via `gips_process sentinel2 -p rad-toa -d 2017-01-30 -v4`, using the formula from #205.

Other important changes:

* refactor `process()` into multiple methods; no semantic difference, just makes the code easier to work on.
* proto product optimization - if `ref-toa` already exists, just load that and save some work.
* tile string normalization support in gips engine
* a few little style improvements, mostly line length changes

### CODE REVIEW

File: gips/test/sys/expected/sentinel2.py [hunk 1]
```diff
@@ -111,29 +111,31 @@
...
+\x1b[0m   rad         Surface-leaving radiance                
+   ref         Surface reflectance                     
 """
 }
 
 
 t_project = {
+    'compare_stderr': False,
     'created': {
-        '0': None,
```
Comment (ags-tolson) [inline gips/test/sys/expected/sentinel2.py]: note both red and green values are wrong; `gips_project` seems to generate nothing but NaN values in both this branch and the prior 200 branch.

File: gips/data/sentinel2/sentinel2.py [hunk 2]
```diff
@@ -335,15 +431,98 @@ def read_raw(self):
...
+        radiance_factors = asset_instance.radiance_factors()
+
+        rad_image = gippy.GeoImage(proto_product)
+
+        for i in range(len(rad_image)):
+            color = rad_image[i].Description()
+            rf = radiance_factors[colors.index(color)]
+            self._time_report(
+                'TOA radiance conversion factor for {} (band {}): {}'.format(color, i + 1, rf))
+            rad_image[i] *= rf
```
Comment (ircwaves) [inline gips/data/sentinel2/sentinel2.py]: Looks like `GeoRaster operator*=(double val)` doesn't exist and swig doesn't barf on its absence.

**Comment by ircwaves** [comment]
Also, I should have caught this on the last PR, but pull the durham.qpj file out of the repo.  it is unnecessary.

File: gips/test/sys/expected/sentinel2.py (hunk 1 above)
Comment (ags-tolson) [inline gips/test/sys/expected/sentinel2.py]: fixed in 231a634 see below.

**Comment by ags-tolson** [comment]
this branch has been superseded; closing unmerged.

**Pull request closed without merging**